File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
def is_correct(s):
    stk = []
    
    for i in range(len(s)):
        if len(stk) == 0:
            stk.append(s[i])
            continue
        
        if stk[-1] == '[' and s[i] == ']':
            stk.pop()
        elif stk[-1] == '(' and s[i] == ')':
            stk.pop()
        elif stk[-1] == '{' and s[i] == '}':
            stk.pop()
        else:
            stk.append(s[i])
    
    if len(stk) == 0:
        return True
    else:
        return False

def solution(s):
    answer = 0
    cnt = 0
    while cnt < len(s):
        if is_correct(s):
            answer += 1
        s = s[1:] + s[0]
        cnt += 1
    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        int len = s.length();
        for(int i = 0; i < len; i++){
            if(isCorrect(s.substring(i, len) + s.substring(0, i))){
                answer++;
            }
        }
        return answer;
    }
    public boolean isCorrect(String s){
        Stack<Character> stk = new Stack<>();
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '[' || s.charAt(i) == '(' || s.charAt(i) == '{'){
                stk.push(s.charAt(i));
            }else{
                if(stk.isEmpty()){
                    return false;
                }
                
                if(s.charAt(i) == ']' && stk.peek() != '['){
                    return false;
                }
                
                if(s.charAt(i) == '}' && stk.peek() != '{'){
                    return false;
                }
                
                if(s.charAt(i) == ')' && stk.peek() != '('){
                    return false;
                }
                
                stk.pop();
            }
        }
        
        if(!stk.isEmpty()){
            return false;
        }
        return true;
    }
}